tests/providers/amazon/aws/triggers/test_base_trigger.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from unittest import mock
+from unittest.mock import MagicMock
+
+import pytest
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsGenericHook
+from airflow.providers.amazon.aws.triggers.base import AwsBaseWaiterTrigger
+from airflow.triggers.base import TriggerEvent
+
+
+class TestImplem(AwsBaseWaiterTrigger):
+    """An empty implementation that allows instantiation for tests."""
+
+    __test__ = False
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def hook(self) -> AwsGenericHook:
+        return MagicMock()
+
+
+class TestAwsBaseWaiterTrigger:
+    def setup_method(self):
+        self.trigger = TestImplem(
+            serialized_fields={},
+            waiter_name="",
+            waiter_args={},
+            failure_message="",
+            status_message="",
+            status_queries=[],
+            return_value=None,
+            waiter_delay=0,
+            waiter_max_attempts=0,
+            aws_conn_id="",
+        )
+
+    def test_region_serialized(self):
+        self.trigger.region_name = "my_region"
+        _, args = self.trigger.serialize()
+
+        assert "region_name" in args
+        assert args["region_name"] == "my_region"
+
+    def test_region_not_serialized_if_omitted(self):
+        _, args = self.trigger.serialize()
+
+        assert "region_name" not in args
+
+    def test_serialize_extra_fields(self):
+        self.trigger.serialized_fields = {"foo": "bar", "foz": "baz"}
+
+        _, args = self.trigger.serialize()
+
+        assert "foo" in args
+        assert args["foo"] == "bar"
+        assert "foz" in args
+        assert args["foz"] == "baz"
+
+    @pytest.mark.asyncio
```

Comment:
What about testing things like multiple attempts, attempts exceeded etc? I suppose those are covered in [`test_waiter_with_logging.py`](https://github.com/apache/airflow/blob/main/tests/providers/amazon/aws/utils/test_waiter_with_logging.py) so they don't need to be repeated.

Reply:
yes indeed, this code is not in the base trigger, so it doesn't have to be tested here imho